plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoPrivileges.java

```diff
@@ -0,0 +1,133 @@
+    private static final String TEST_COLLECTION = "testCollection";
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        MongoServer mongoServer = closeAfterClass(setupMongoServer());
+        return createMongoQueryRunner(mongoServer.getConnectionString(TEST_USER, TEST_PASSWORD).toString() + "/?authSource=" + TEST_DATABASE);
+    }
+
+    @Test
+    public void testSchemasVisibility()
+    {
+        assertThat(getQueryRunner().execute("SHOW SCHEMAS FROM mongodb")
+                .getOnlyColumnAsSet())
+                .containsExactly("information_schema", TEST_DATABASE);
+    }
+
+    @Test
+    public void testTablesVisibility()
+    {
+        assertThat(getQueryRunner().execute("SHOW TABLES FROM mongodb." + TEST_DATABASE)
+                .getOnlyColumnAsSet())
+                .containsExactly(TEST_COLLECTION.toLowerCase(Locale.ENGLISH));
+    }
+
+    private static MongoServer setupMongoServer()
+    {
+        MongoServer server = new MongoServer("4.4.17", ImmutableMap.of(
+                "MONGO_INITDB_ROOT_USERNAME", ROOT_USER,
+                "MONGO_INITDB_ROOT_PASSWORD", ROOT_PASSWORD,
+                "MONGO_INITDB_DATABASE", "admin"));
+        try (MongoClient client = MongoClients.create(server.getConnectionString(ROOT_USER, ROOT_PASSWORD))) {
+            MongoDatabase testDatabase = client.getDatabase(TEST_DATABASE);
+            createTestRole(testDatabase);
+            createTestUser(testDatabase);
+            testDatabase.createCollection("_schema");
+            testDatabase.createCollection(TEST_COLLECTION);
+            testDatabase.createCollection("anotherCollection"); // this collection/table should not be visible
+            client.getDatabase("another").createCollection("_schema"); // this database/schema should not be visible
+        }
+        return server;
+    }
+
+    private static DistributedQueryRunner createMongoQueryRunner(String connectionUrl)
+            throws Exception
+    {
+        DistributedQueryRunner queryRunner = null;
+        try {
+            queryRunner = DistributedQueryRunner.builder(testSessionBuilder()
+                            .setCatalog(Optional.empty())
+                            .setSchema(Optional.empty())
+                            .build())
+                    .build();
+            queryRunner.installPlugin(new MongoPlugin());
+            queryRunner.createCatalog("mongodb", "mongodb", ImmutableMap.of(
+                    "mongodb.case-insensitive-name-matching", "true",
+                    "mongodb.connection-url", connectionUrl));
+            return queryRunner;
+        }
+        catch (Throwable e) {
+            closeAllSuppress(e, queryRunner);
+            throw e;
+        }
+    }
+
+    private static void createTestRole(MongoDatabase database)
+    {
+        Double commandStatus = database.runCommand(new BsonDocument("createRole", new BsonString(TEST_ROLE))
```

Comment:
Is it possible to use `Document` class to improve readability?

Reply:
Got it, I also try to improve readability by spiting it into few methods